src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -105,6 +105,121 @@
+                    <!-- if the categorization is 'MUTATION_TYPE' (table) -->
+                    <if test="mutationDataFilter.categorization == @org.cbioportal.web.parameter.MutationOption@MUTATION_TYPE">
+                        <foreach item="dataFilterValues" collection="mutationDataFilter.values" separator="INTERSECT">
+                            SELECT DISTINCT sample_unique_id
+                            FROM genomic_event_derived
+                            WHERE hugo_gene_symbol = #{mutationDataFilter.hugoGeneSymbol}
+                                AND variant_type = 'mutation'
+                                AND mutation_type IN (
+                                <foreach item="dataFilterValue" collection="dataFilterValues" separator=",">
+                                    #{dataFilterValue.value}
+                                </foreach>
+                                )
+                        </foreach>
+                    </if>
+                </foreach>
+            </if>
+            <if test="studyViewFilter.genomicDataFilters != null and !studyViewFilter.genomicDataFilters.isEmpty()">
+                <foreach item="genomicDataFilter" collection="studyViewFilter.genomicDataFilters" open="INTERSECT" separator="INTERSECT">
+                    <bind name="valueColumn" value="genomicDataFilter.profileType + '_value'" />
+                    <!-- 1. check if 'NA' is selected -->
+                    <bind name="containsNA" value="false" />
+                    <foreach item="dataFilterValue" collection="genomicDataFilter.values">
+                        <if test="dataFilterValue.value == 'NA'">
+                            <bind name="containsNA" value="true" />
+                        </if>
+                    </foreach>
+                    WITH cna_query AS (
+                        SELECT sample_unique_id, ${valueColumn}
+                        FROM genetic_alteration_derived_cna
+                        WHERE hugo_gene_symbol = #{genomicDataFilter.hugoGeneSymbol}
+                            AND cancer_study_identifier IN
+                            <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                                #{studyId}
+                            </foreach>
+                    )
+                    SELECT DISTINCT sd.sample_unique_id
+                    FROM sample_derived sd
+                        LEFT JOIN cna_query ON sd.sample_unique_id = cna_query.sample_unique_id
+                    WHERE cancer_study_identifier IN
+                        <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                            #{studyId}
+                        </foreach>
+                        <choose>
+                            <!-- 2. if no 'NA' is selected -->
+                            <when test="!containsNA">
+                                AND ${valueColumn} IN
```

Comment:
should valueColumn still be dynamic given that we know this cna?

Reply:
It could also be 'gistic', which is another profile of CNA as I understand.
Maybe this is the level of generic they're thinking of? That I can't be sure.

Reply:
i see now that we've broken out values into mulitple columns in the the sling_db_2024_07_19.genetic_alteration_derived_cna table.  i'm curious how we did that and why.